### PR TITLE
Mix and Erlang.mk support

### PR DESCRIPTION
**Pull request by ericbmerritt**

### CODE REVIEW

File: src/h2n_resolver.erl [hunk 1]
```diff
@@ -2,7 +2,7 @@
 %%% @doc
 %%%
 %%% This module resolves the dependencies. It is *not* a full
-%%% constraint solver. It does *not* make sure that everything meets
```
Comment (gleber) [inline src/h2n_resolver.erl]: Looks like one line got lost

File: src/hex2nix.erl [hunk 2]
```diff
@@ -221,20 +208,25 @@ filter_by_name(Filter, App) ->
...
-
--spec simplify_deps([[any()]]) ->
+-spec simplify_and_remove_duplicate_deps([[any()]]) ->
                            [app()].
-simplify_deps(Deps) ->
-    lists:map(fun simplify_dep/1, Deps).
+simplify_and_remove_duplicate_deps(Deps) ->
+    lists:foldl(fun (El0, Acc) ->
+                        El1 = {Name, _} = simplify_dep(El0),
+                        case Name of
```
Comment (gleber) [inline src/hex2nix.erl]: Looks like a debug statemept

File: src/h2n_resolver.erl [hunk 3]
```diff
@@ -74,6 +75,13 @@ expand_dependencies(Deps = #indexed_deps{detail = Detail}
     case dict:find(App, Detail) of
         {ok, AppDeps} ->
             DepsList = update_deps_list(Deps, AppDeps),
+            case App of
+                {<<"postgrex">>, <<"0.11.1">>} ->
```
Comment (gleber) [inline src/h2n_resolver.erl]: One more debug string

File: src/h2n_generate.erl [hunk 4]
```diff
@@ -76,15 +77,34 @@ format_position(_, _, _) ->
...
-    par([break(sep([text("buildRebar3"), text("{")]))
-        , nest(par([src(Dep)
+app_body(Dep = #dep_desc{app = {Name, Vsn}
+                        , build_tool = rebar3}, Deps) ->
+    par([break(sep([text("buildRebar3"), text("({")]))
+        , nest(par([key_value(<<"name">>, Name)
+                   , key_value(<<"version">>, Vsn)
+                   , src(Dep)
                    , format_compile_port(Dep)
-                   , build_plugins(BuildPlugins)
```
Comment (gleber) [inline src/h2n_generate.erl]: Why is support for buildPlugins removed?

Comment (sunlight-algorithmic-user) [inline src/h2n_generate.erl]: this got commented out before I pulled in your stuff. I will add it back

**Comment by gleber** [comment]
Looks good!

**Comment by ericbmerritt** [comment]
There was a build bug. I will get it fixed and produce another commit.

**Pull request merged**